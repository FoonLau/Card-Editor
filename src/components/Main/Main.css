.app-main {
  display: flex;
  flex-wrap: wrap;
}

.app-main-editor,
.app-main-output {
  width: 50%;
}

.app-main-editor {
  border-right: 1px solid darkgray;
  box-sizing: border-box;
}

.app-main-gallery {
  width: 100%;
  border-top: 1px solid darkgray;
}