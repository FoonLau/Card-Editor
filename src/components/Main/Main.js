import React from 'react';
import Editor from './Editor/Editor';
import Gallery from './Gallery/Gallery';
import Output from './Output/Output';

export default () => (
  <div className="app-main">
    <div className="app-main-editor">
      <Editor />
    </div>
    <div className="app-main-output">
      <Output />
    </div>
    <div className="app-main-gallery">
      <Gallery />
    </div>
  </div>
);
