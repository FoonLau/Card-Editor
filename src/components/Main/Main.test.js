import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Main from './Main';

const store = configureMockStore()({
  edittingItem: {},
  items: []
});

it('should render correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Main />
    </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot(tree);
});
