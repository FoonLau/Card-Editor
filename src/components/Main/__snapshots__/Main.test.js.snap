// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div
  className="app-main"
>
  <div
    className="app-main-editor"
  >
    <div
      className="app-editor"
    >
      <div
        className="app-editor-item"
      >
        <label
          htmlFor="app-editor-background"
        >
          background colour
        </label>
        <input
          id="app-editor-background"
          onChange={[Function]}
          type="color"
          value={undefined}
        />
      </div>
      <div
        className="app-editor-item"
      >
        <label
          htmlFor="app-editor-width"
        >
          width
        </label>
        <input
          id="app-editor-width"
          onChange={[Function]}
          type="number"
          value={undefined}
        />
      </div>
      <div
        className="app-editor-item"
      >
        <label
          htmlFor="app-editor-height"
        >
          height
        </label>
        <input
          id="app-editor-height"
          onChange={[Function]}
          type="number"
          value={undefined}
        />
      </div>
      <div
        className="app-editor-item"
      >
        <label
          htmlFor="app-editor-border-radius"
        >
          border radius
        </label>
        <input
          id="app-editor-border-radius"
          onChange={[Function]}
          type="range"
          value={undefined}
        />
      </div>
      <button
        className="app-editor-save"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="app-main-output"
  >
    <div
      className="app-output"
    >
      <div
        className="app-item"
        style={
          Object {
            "borderRadius": "undefined%",
            "dispatch": [Function],
          }
        }
      />
    </div>
  </div>
  <div
    className="app-main-gallery"
  >
    <div
      className="app-gallery"
    />
  </div>
</div>
`;
